### PR TITLE
Add Public Key Pins Filter

### PR DESCRIPTION
**Pull request by eric-gonzales**
* Add HPKP filter
* Fix minor bug with X-XSS Filter

### CODE REVIEW

File: src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/XFrameOptionsGrailsPlugin.groovy [hunk 1]
```diff
@@ -1,17 +1,7 @@
...
-import com.mrhaki.grails.plugin.xframeoptions.config.XContentTypeOptionsConfiguration
-import com.mrhaki.grails.plugin.xframeoptions.config.XFrameOptionsConfiguration
-import com.mrhaki.grails.plugin.xframeoptions.config.XXSSProtectionConfiguration
-import com.mrhaki.grails.plugin.xframeoptions.web.ContentSecurityPolicyFilter
-import com.mrhaki.grails.plugin.xframeoptions.web.ReferrerPolicyFilter
-import com.mrhaki.grails.plugin.xframeoptions.web.StrictTransportSecurityFilter
-import com.mrhaki.grails.plugin.xframeoptions.web.XContentTypeOptionsFilter
-import com.mrhaki.grails.plugin.xframeoptions.web.XFrameOptionsFilter
-import com.mrhaki.grails.plugin.xframeoptions.web.XXSSProtectionFilter
+import com.mrhaki.grails.plugin.xframeoptions.config.*
```
Comment (dbpfindexp) [inline src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/XFrameOptionsGrailsPlugin.groovy]: 👍

File: src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/PublicKeyPinsConfiguration.groovy [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        final String prefix = 'grails.plugin.xframeoptions.publickeypins'
+        enabled = config.getProperty("${prefix}.enabled", Boolean, DEFAULT_ENABLED)
+        subdomains = config.getProperty("${prefix}.subdomains", Boolean, DEFAULT_SUBDOMAINS)
+        maxAge = config.getProperty("${prefix}.maxage", String, DEFAULT_MAX_AGE)
+        spkiFingerprint = config.getProperty("${prefix}.spkiFingerprint", String, DEFAULT_SPKI_FINGERPRINT)
+        spkiSecondaryFingerprint = config.getProperty("${prefix}.spkiSecondaryFingerprint", String, DEFAULT_SPKI_SECONDARY_FINGERPRINT)
+    }
+
+    boolean isFilterEnabled() {
+       enabled
```
Comment (dbpfindexp) [inline src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/PublicKeyPinsConfiguration.groovy]: can we add the `return` here

Comment (eric-gonzales) [reply src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/PublicKeyPinsConfiguration.groovy]: on it!

Comment (eric-gonzales) [reply src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/PublicKeyPinsConfiguration.groovy]: done

File: src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/PublicKeyPinsConfiguration.groovy [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        spkiFingerprint = config.getProperty("${prefix}.spkiFingerprint", String, DEFAULT_SPKI_FINGERPRINT)
+        spkiSecondaryFingerprint = config.getProperty("${prefix}.spkiSecondaryFingerprint", String, DEFAULT_SPKI_SECONDARY_FINGERPRINT)
+    }
+
+    boolean isFilterEnabled() {
+        return enabled
+    }
+
+    private boolean hasCustomSettings() {
+        spkiSecondaryFingerprint != '' || subdomains || preload
```
Comment (Joshua-Brent-Collins) [inline src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/PublicKeyPinsConfiguration.groovy]: If we want to be explicit with returns lets add it here and in is filter enabled in (XXSSProtectionConfiguration.groovy) then I'm good.

Comment (eric-gonzales) [reply src/main/groovy/com/mrhaki/grails/plugin/xframeoptions/config/PublicKeyPinsConfiguration.groovy]: can do

**Pull request merged**